src/api/spec/policies/assignment_policy_spec.rb

```diff
@@ -0,0 +1,47 @@
+RSpec.describe AssignmentPolicy do
+  subject { described_class }
+
+  permissions :create? do
+    before do
+      Flipper.enable(:foster_collaboration)
+    end
+
+    context 'a package collaborator' do
+      let(:package_collaborator) { package.maintainers.first }
+      let(:package) { create(:package_with_maintainer) }
+      let(:assignment) { create(:assignment, assigner: package_collaborator, assignee: package_collaborator, package: package) }
+
+      it { is_expected.to permit(package_collaborator, assignment) }
+    end
+
+    context 'a project collaborator' do
```

Comment:
Don't test the different types of users for `is_assignee_a_collaborator?` through the AssignmentPolicy please. Test it in the `Assignment` spec.

Reply:
what about now?